app/javascript/src/components/Dashboard/Notes/constants.js

```diff
@@ -1,13 +1,62 @@
 import * as yup from "yup";
 
+import { buildSelectOption } from "utils/index";
+
+export const ROLE_DATA = [
+  "UX Designer",
+  "Tester",
+  "Analyst",
+  "Back-end developer",
+  "Front-end developer",
+];
+
+export const TAGS_DATA = [
+  "Getting Started",
+  "Onboarding",
+  "User Flow",
+  "UX",
+  "Bugs",
+  "V2",
+];
+
 export const NOTES_FORM_INITIAL_FORM_VALUES = {
   title: "",
   description: "",
+  role: null,
+  tags: [],
 };
 
 export const NOTES_FORM_VALIDATION_SCHEMA = yup.object().shape({
   title: yup.string().required("Title is required"),
   description: yup.string().required("Description is required"),
+  role: yup
+    .object()
+    .nullable()
+    .shape({
+      label: yup
+        .string()
+        .oneOf(buildSelectOption(ROLE_DATA).map(role => role.label)),
+      value: yup
+        .string()
+        .oneOf(buildSelectOption(ROLE_DATA).map(role => role.value)),
+    })
+    .required("Role is required"),
+  tags: yup
+    .array(
+      yup
+        .object()
+        .nullable()
+        .shape({
+          label: yup
+            .string()
+            .oneOf(buildSelectOption(TAGS_DATA).map(tag => tag.label)),
+          value: yup
+            .string()
+            .oneOf(buildSelectOption(TAGS_DATA).map(tag => tag.value)),
+        })
+    )
+    .min(1, "At least one tag is required")
+    .required("Tag is required"),
```

Comment:
Can you find a way to avoid invoking `buildSelectOption(TAGS_DATA)` or `buildSelectOption(ROLE_DATA)` multiple times within this scope and still return the proper validation object?

Reply:
I fixed this issue by calling the `buildSelectOption` at the time of initializing the `tags` and `role` data.
@yedhink  I fixed the above-mentioned mistakes in the next PR.